src/pages/admin/problems/add.page.tsx

```diff
@@ -0,0 +1,200 @@
+import React from "react"
+import "@uiw/react-md-editor/markdown-editor.css"
+import "@uiw/react-markdown-preview/markdown.css"
+import "katex/dist/katex.css"
+import dynamic from "next/dynamic"
+import katex from "katex"
+import {
+   Box,
+   DescriptionTextField,
+   TitleTextField,
+   HorizaontalBox,
+   Card,
+   TextField,
+} from "./styles"
+import { useServerManager } from "../../../api/server"
+import { Button, CardContent, CardHeader, Divider, Typography } from "@mui/material"
+import { useForm, Controller } from "react-hook-form"
+
+const MDEditor = dynamic(() => import("@uiw/react-md-editor"), { ssr: false })
+
+export default function AddProblem(): JSX.Element {
+   const serverManager = useServerManager()
+
+   const { handleSubmit, control, getValues } = useForm({
+      defaultValues: {
+         title: "",
+         description: "",
+         text: "",
+         maxMemory: "",
+         maxTime: "",
+         //dataSets: [],
+      },
+   })
+
+   const handleUpload = () => {
+      // const data = new FormData()
+      // Object.entries(problemData).forEach((entry) => {
+      //    data.append(entry[0], entry[1])
+      // })
+      // serverManager.createProblem(data)
+      const data = getValues()
+      console.log(data)
+   }
+
+   return (
+      <Box>
+         <Controller
```

Comment:
This logic can be implemented using a self created component to avoid repeat code